### ISSUE TITLE
Login (auth and link) Issues

### ISSUE THREAD

**Issue by sebastianfelipe** [strongloop/loopback-component-passport#179]
Hi everyone, recently I've been implementing this tool, and doing some tests I found that some use cases are rare. Here's the thing:

First case:
- ClassicUser (email + password) link a Facebook account (call it FacebookLink)
- After that, you want to log in with you facebook account, so, you use /auth/facebook
- Between the step before and this step, it doesn't return the ClassicUser information (as it should be), but it creates another account, a second account call it FacebookAuth, where is a new account.

In other words, it doesn't logged in as a FacebookLink account, using all the information about the ClassicAccount, instead it creates a new one, totally different, where we can't use ACLs to the REST API like if it is a Classic one.

Second case:
- Suppose you have a FacebookAuth (account first sign up with facebook), and then you create a ClassicAccount (email + password).
- You log in with your ClassicAccount and decide to link your facebook account (that in this moment is in use). When you try to link it, you can do it... I think is very weird. It is a very weird use case.

Anyway, that wouldn't weird if there would be a possibility to log in with precedence in linked accounts firstly. Or maybe just determine a way where we can log in using the linked accounts, because the linked accounts can't be useful to log in, only to take information.

Another solution is create a new way to log in, that is, replacing the callback in use. But it still be weird, because if you link an account to another one, you shouldn't be able to log in with that. And also, you can create another account instead (second case). It shouldn't let the creation of a new account if it is already created.

I hope receive some feedback or at least a way to deal with it in a "clean" way.

Thanks,
Sebastián.

**Comment by danielmihai** [comment]
I subscribe to the same issue. I have the same problems, couldn't say it better myself :)
:+1:  @sebastianfelipe

**Comment by ZepAviator** [comment]
Did you ever come to a solution. I'd like to route all users into a common email. So, if you log in through facebook who's profile info has someuser@somedomain.com and then you create and validate an email that is someuser@somedomain.com - they both end up linked as the same account automatically.

**Comment by danielmihai** [comment]
I modified the component myself. I keep it in github knowing that it's not a good practice. take a look at the loopback example from my account if you like to. Did this a while ago so I don't really remember what flow I implemented.

**Comment by smith64fx** [comment]
Really interested in this issue.

That's very important.

@danielmihai Why not go for a pull request?

**Comment by danielmihai** [comment]
@smith64fx Did it work? 

I forked the official repository now and I've seen it diverged a lot (as it normally would do so). I tried preparing for a pull request moments ago. I merged my changes in the current loopback-2.x branch. I have to fix some parts again, because the unit tests fail. I'll investigate further when I have/make the time. 

What's your status @smith64fx ? keep in touch

Thanks,
Daniel.
